### PR TITLE
Added Istio Unit Tests

### PR DESCRIPTION
**Pull request by mattmahoneyrh**

### CODE REVIEW

File: ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml [hunk 1]
```diff
@@ -96,6 +96,55 @@ tests:
...
+          memory: 100Mi
+      timeout: 20m0s
+    - as: maistra-istio-unit-3-0
+      cli: latest
+      commands: |
+        oc rsh -n "${MAISTRA_NAMESPACE}" "${MAISTRA_SC_POD}" \
+          bash -c \
+          "cd /work;
+          entrypoint \
+          make -e "T=-v -count=1" build racetest binaries-test"
```
Comment (yxun) [inline ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml:135]:
```suggestion
          make -e 'T=-v -count=1' build racetest binaries-test"
```

File: ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml [hunk 2]
```diff
@@ -96,6 +96,55 @@ tests:
...
+        oc cp "${MAISTRA_NAMESPACE}"/"${MAISTRA_SC_POD}":"${ARTIFACT_DIR}"/. "${ARTIFACT_DIR}"
+      env:
+      - name: BUILD_WITH_CONTAINER
+      - name: MAISTRA_NAMESPACE
+      - name: MAISTRA_SC_POD
+      from: src-test
+      resources:
+        requests:
+          cpu: "2"
+          memory: 2Gi
```
Comment (yxun) [inline ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml:145]:
```suggestion
          cpu: 100m
          memory: 100Mi
```
This section only runs `oc rsh`  100m resource is enough. The real test execution runs in the remote pod and resources are defined as the node type  COMPUTE_NODE_TYPE: m5d.4xlarge above.

File: ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml [hunk 3]
```diff
@@ -96,6 +96,55 @@ tests:
...
+      timeout: 1h0m0s
```
Comment (yxun) [inline ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml:146]:
```suggestion
      timeout: 2h0m0s
```

Increase timeout value for the additional cluster provisioning time.

File: ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml [hunk 4]
```diff
@@ -96,6 +96,55 @@ tests:
...
+    workflow: servicemesh-istio-e2e-aws
```
[the remark this answers is not in the document]
Comment (yxun) [reply ci-operator/config/maistra/istio/maistra-istio-maistra-3.0.yaml:147]: You also need to run `make update` for generating the ci-operator/jobs/maitra/ file changes. And then this PR can include those changes.

**Comment by mattmahoneyrh** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-3.0-maistra-istio-unit-3-0

**Comment by mattmahoneyrh** [comment]
Appears to be a provisioning failure, looking at the install pod log.
[Install.log](https://github.com/openshift/release/files/11564897/Install.log)

**Comment by yxun** [comment]
> Appears to be a provisioning failure, looking at the install pod log. [Install.log](https://github.com/openshift/release/files/11564897/Install.log)

right, that's a known provision error. I will retry it

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-maistra-3.0-maistra-istio-unit-3-0

**Comment by mattmahoneyrh** [comment]
Unit tests are now running and passing. There are 'skipped' tests with this msg "
Reason: util.go:82: not enough bytes to create int". Is this another known issue with unit tests?

**Comment by yxun** [comment]
> Unit tests are now running and passing. There are 'skipped' tests with this msg " Reason: util.go:82: not enough bytes to create int".
> 
> @yxun Is this known issue with unit tests?

All good. Thanks, Matt.
I will approve this PR and also add /pj-rehearse  ack label

**Pull request merged**